lms/djangoapps/courseware/tests/test_password_history.py

```diff
@@ -338,3 +342,29 @@ def test_password_policy_on_password_reset(self):
         }, follow=True)
 
         self.assertIn(success_msg, resp.content)
+
+    @ddt.data(
+        ('foo', 'foobar'),
+        ('', ''),
+    )
+    @ddt.unpack
+    def test_password_reset_form_invalid(self, password1, password2):
+        """
+        Tests that password rest fail when providing bad passwords and error message is displayed
+        to the user.
+        """
+        user_email, _ = self._setup_user()
+        err_msg = 'Password reset form did not validate, please try again'
+
+        # try to reset password, it should fail
+        user = User.objects.get(email=user_email)
+        token = default_token_generator.make_token(user)
+        uidb36 = int_to_base36(user.id)
+
+        # try to do a password reset with the same password as before
+        resp = self.client.post('/password_reset_confirm/{0}-{1}/'.format(uidb36, token), {
+            'new_password1': password1,
+            'new_password2': password2,
+        }, follow=True)
+        self.assertPasswordResetError(resp, err_msg, True)
```

Comment:
Nit: for code readability, add `valid_link=` when passing `True` so the reader knows the semantic of the boolean value.